src/Http/WebUtilities/src/MultipartReader.cs

```diff
@@ -51,6 +52,8 @@ public MultipartReader(string boundary, Stream stream, int bufferSize)
             throw new ArgumentNullException(nameof(boundary));
         }
 
+        boundary = HeaderUtilities.RemoveQuotes(new StringSegment(boundary)).ToString();
```

Comment:
Style: Move this down to where _boundary is being initialized.

Reply:
So just put it right above that line? Yeah I can do that.

Reply:
Does this allocate when there's nothing to do?

Reply:
Are you talking about that fact that RemoveQuotes is called before all of the null checks? I just pushed up a change so that it's now being called right before the boundary variable is actually used, so it shouldn't be allocated unless it's used now.

Reply:
It doesn't allocate.
https://github.com/dotnet/aspnetcore/blob/bcdb13d1f2e42f2d318ead75e5e86fc27c78a117/src/Http/Headers/src/HeaderUtilities.cs#L570-L574

Reply:
Have you checked if there is no string allocation when calling ToString() when double quotes are not present in an input boundary? What about checking if there are double quotes first and only after that to remove them and call ToString()?

Reply:
I checked, there's no allocation since this ends up returning the whole original string.
https://github.com/dotnet/runtime/blob/2ea67f1f42e8170676c9e9591a974e39513390db/src/libraries/Microsoft.Extensions.Primitives/src/StringSegment.cs#L661
https://github.com/dotnet/runtime/blob/2ea67f1f42e8170676c9e9591a974e39513390db/src/libraries/Microsoft.Extensions.Primitives/src/StringSegment.cs#L81
https://github.com/dotnet/runtime/blob/215b39abf947da7a40b0cb137eab4bceb24ad3e3/src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs#L1831-L1834

Reply:
Thanks for the clarification